src/test/java/com/spotify/docker/client/DockerCertificatesTest.java

```diff
@@ -0,0 +1,149 @@
+  private ArgumentCaptor<char[]> password = ArgumentCaptor.forClass(char[].class);
+
+  @Test(expected = DockerCertificateException.class)
+  public void testBadDockerCertificates() throws Exception {
+    // try building a DockerCertificates with specifying a cert path to something that
+    // isn't a cert
+    DockerCertificates.builder()
+        .dockerCertPath(getResourceFile("dockerInvalidSslDirectory"))
+        .build();
+  }
+
+  @Test
+  public void testNoDockerCertificatesInDir() throws Exception {
+    final Path certDir = Paths.get(System.getProperty("java.io.tmpdir"));
+    final Optional<DockerCertificatesStore> result = DockerCertificates.builder()
+        .dockerCertPath(certDir)
+        .build();
+    assertThat(result.isPresent(), is(false));
+  }
+
+  @Test
+  public void testDefaultDockerCertificates() throws Exception {
+    DockerCertificates.builder()
+        .dockerCertPath(getCertPath())
+        .sslFactory(factory)
+        .build();
+
+    verify(factory).newSslContext(keyStore.capture(), password.capture(), trustStore.capture());
+
+    final KeyStore.PrivateKeyEntry pkEntry = (KeyStore.PrivateKeyEntry) keyStore.getValue()
+        .getEntry("key", new KeyStore.PasswordProtection(password.getValue()));
+
+    final KeyStore caKeyStore = trustStore.getValue();
+
+    assertNotNull(pkEntry);
+    assertNotNull(pkEntry.getCertificate());
+    assertNotNull(caKeyStore.getCertificate("o=boot2docker"));
+  }
+
+
+  @Test
+  public void testDockerCertificatesWithMultiCa() throws Exception {
+    DockerCertificates.builder()
+        .dockerCertPath(getCertPath())
+        .caCertPath(getVariant("ca-multi.pem"))
+        .sslFactory(factory)
+        .build();
+
+    verify(factory).newSslContext(keyStore.capture(), password.capture(), trustStore.capture());
+
+    final KeyStore.PrivateKeyEntry pkEntry = (KeyStore.PrivateKeyEntry) keyStore.getValue()
+        .getEntry("key", new KeyStore.PasswordProtection(password.getValue()));
+
+    assertNotNull(pkEntry);
+    assertNotNull(pkEntry.getCertificate());
+    assertNotNull(trustStore.getValue().getCertificate(
+        "cn=ca-test,o=internet widgits pty ltd,st=some-state,c=cr"));
+    assertNotNull(trustStore.getValue().getCertificate(
+        "cn=ca-test-2,o=internet widgits pty ltd,st=some-state,c=cr"));
+  }
+
+  @Test
+  public void testReadPrivateKeyPkcs1() throws Exception {
+    DockerCertificates.builder()
+        .dockerCertPath(getCertPath())
+        .clientKeyPath(getVariant("key-pkcs1.pem"))
+        .sslFactory(factory)
+        .build();
+
+    verify(factory).newSslContext(keyStore.capture(), password.capture(), trustStore.capture());
+
+    final KeyStore.PrivateKeyEntry pkEntry = (KeyStore.PrivateKeyEntry) keyStore.getValue()
+        .getEntry("key", new KeyStore.PasswordProtection(password.getValue()));
+
+    assertNotNull(pkEntry.getPrivateKey());
+  }
+
+  @Test
+  public void loadReadPrivateKeyPkcs8() throws Exception {
```

Comment:
Yes, you right. It is wrong here. I can replace it, but I would like to know how do you prefer committing the fix. I personally don't like "dangling" commits with oneline fixes and I usually squash it and use "force", but then I think all approvals will disappear and you will have to review whole commit again.